src/test/groovy/com/github/j2objccontrib/j2objcgradle/tasks/XcodeTaskTest.groovy

```
@@ -1107,6 +1112,43 @@ class XcodeTaskTest {
     }
 
     @Test
+    void testUpdatePodMethods_customConfigurations() {
+        XcodeTask.PodspecDetails podspecDetails =
+                new XcodeTask.PodspecDetails(
+                        'PROJ',
+                        new File('/SRC/PROJ/BUILD/j2objc-PROJ-debug.podspec'),
+                        new File('/SRC/PROJ/BUILD/j2objc-PROJ-release.podspec'),
+                        ['Debug', 'Beta'], ['Release', 'Preview'])
+        List<String> podMethodLines = XcodeTask.podMethodLines(
+                podspecDetails,
+                new File('/SRC/ios/Podfile'))
+        List<String> expectedLines = [
+                "def j2objc_PROJ",
+                "    pod 'j2objc-PROJ-debug', :configuration => ['Debug','Beta'], :path => '../PROJ/BUILD'",
+                "    pod 'j2objc-PROJ-release', :configuration => ['Release','Preview'], :path => '../PROJ/BUILD'",
+                "end"]
+        assert expectedLines == podMethodLines
+
+        // Empty configurations list should omit pod line, while an empty string should be included
```

If it's an error, then you should throw an exception and then catch it in test. See this example elsewhere in the file:

    expectedException.expect(InvalidUserDataException.class)